academic-services/knowledge-exploration-service/get-started-create-api-instances.md

```diff
@@ -5,24 +5,79 @@ ms.topic: tutorial
+
+Each MAKES deployment includes the scripts required to provision an instance of MAKES in Azure.  These tools are located in the 'tools' folder that you verified above.  To download the scripts, select the 'kesm.zip' file from Storage Explorer and click 'Download' from the top ribbon:
+
+![Download kesm.zip](media/get-started-download-kesm.png)
+
+Save this file to a local folder.
+
+## Unzip the script package
+
+Once the file has been downloaded you will need to extract the contents.  Right click on the kesm.zip file.  If you are using the Windows operating system select "Extract All" from the context menu.  If you are using linux, select "Extract Here" or "Unzip Here".  If you are using a Mac, double-click the zip file and it will decompress into the same folder.
+
+If this has completed successfully you will have extracted a single file named kesm.exe.
+
+## Run the script
+
+At this point to are ready to deploy an instance of MAKES to your Azure account.  Open a command prompt (Windows) or terminal window (Mac) and navigate to the folder that you extracted the kesm.exe file to.
 
 Execute the following command to deploy hosting resources
 
 ```cmd
-makes.exe CreateHostResource --MakesPackageBaseDir "https://[storage account name].blob.core.windows.net/makes/[release version]/"
+kesm.exe DeployHost --HostName [whatever_name_you_would_like] --MakesPackage "https://[your_storage_account_name].blob.core.windows.net/makes/[release version]/"
+```
+
+- Replace "[whatever_name_you_would_like]" with the host name for your service.  The hostname will be the host name of the server where your MAKES deployment will be hosted.  Ex: If you used 'testmakes01102020', your MAKES API will be hosted at http://testmakes10012020.westus.cloudapp.azure.net
+
+- Replace "[your_storage_account_name]" with the name of the storage account you downloaded the scripts from above.
+
+At this point the tool will take care of creating all of the required resources and deploying your instance of MAKES.  This process can take a long time.  During the initial run it can take up to 3 hours to complete.  During the first run the script generates some resources that you only need to create once.  Subsequent deployments will not take as long if you specify existing resources.  See the [Command line Reference](reference-makes-command-line-tool.md) for more details.  
+
+Depending on the region your storage account is in and the region you are deploying to, deployment may take longer as the default indexes are quite large and need to be copied.  By default, the tool deploys to the WestUS region of Azure.  You can change this by adding parameters to the script.  For a reference of all the available parameters type:
+
+```cmd
+kesm.exe --help
+
+// For command specific help; type the command, appending '--help'
+
+kesm.exe DeployHost --help
 ```
 
-## Deploy MAKES API instances
+Or, you can visit the [kesm.exe Command Line Reference](reference-makes-command-line-tool.md).
 
```

Comment:
When I was using the tool I made a stupid mistake by using a hyphen in my host name. This caused me to waste 30 minutes as the VM template was created but then deployment immediately failed as host names can't include hyphens.

We should probably add an "[!IMPORTANT]" section above that warns users of this pitfall (or modify the tool to make this check up-front)

Reply:
I actually had Alvin change the validation on the parameters to happen at initial execution.  There is a PR out for this change.  I ran into this as well :|.

Reply:
Excellent!@

---
In reply to: [370872227](https://github.com/MicrosoftDocs/microsoft-academic-services/pull/90#discussion_r370872227) [](ancestors = 370872227)